src-tauri/build.rs

```diff
@@ -1,3 +1,97 @@
+use eyre::{bail, Result};
+use flate2::read::GzDecoder;
+use reqwest::get;
+use std::env;
+use std::env::consts::{ARCH, OS};
+use std::fs;
+use std::fs::File;
+use std::io::copy;
+use std::io::BufReader;
+use std::io::Read;
+use tar::Archive;
+
 fn main() {
+    tauri::async_runtime::block_on(setup_binary()).unwrap();
     tauri_build::build()
 }
+
+pub async fn setup_binary() -> Result<()> {
+    let target = determine_target();
+    let binary_name = "meroctl";
+    let cache_dir = std::env::temp_dir().join("meroctl");
+    std::fs::create_dir_all(&cache_dir).expect("Failed to create cache directory");
+
+    let url = format!(
+        "https://github.com/calimero-network/core/releases/latest/download/{}.tar.gz",
```

Comment:
we can just get the target from the env, and pass it directly since it's just `meroctl_<target>`

```suggestion
        "https://github.com/calimero-network/core/releases/latest/download/meroctl_{}.tar.gz",
```